Odata-docs/webapi/model-builder-revisions-capabilities.md

```diff
@@ -0,0 +1,177 @@
+{
+  public int CustomerId { get; set; }
+  public string Name { get; set; }
+}
+
+public class Address
+{
+    public string City {get;set;}
+}
+
+public enum Color
+{
+    Red,
+    Green,
+    Blue
+}
+```
+Revisions can be added to all model elements. So far, the OData Model Builder supports revisions to the following model elements:
+
+1. EntitySets
+1. Actions and Action Imports
+1. Functions and Function Imports
+1. Entity Types
+1. Complex Types
+1. Enum Types
+1. Enum Members
+1. Properties
+
+There are three kinds of revisions that can be added to any of the model elements above: 
+
+```csharp
+enum RevisionKind
+{
+    //when a model element is added
+    Added,
+
+    //when a model element is modified
+    Modified,
+
+    //when a model elemeny is deprecated
+    Deprecated
+}
+```
+
+
+To add revisions to an entity set for example. use the code below:
+
+```csharp
+var builder = new ODataConventionModelBuilder();
+var customerConfiguration = builder.EntitySet<Customer>("Customers").HasRevisions(a => a.HasVersion("v1.2").HasKind(RevisionKind.Added).HasDescription("Added a new entity set"));
+```
+
+Use this structure to add revisions to all the other supported model elements above. 
+
+$metadata
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+    <edmx:DataServices>
+        <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="Default">
+            <EntityType Name="Customer">
+                <Key>
+                    <PropertyRef Name="CustomerId" />
+                </Key>
+                <Property Name="CustomerId" Type="Edm.Int32" Nullable="false" />
+                <Property Name="Name" Type="Edm.String" />
+            </EntityType>
+            <EntityContainer Name="Container">
+                <EntitySet Name="Customers" EntityType="Default.Customer" />
+            </EntityContainer>
+            <Annotations Target="Default.Container/Customers">
+                <Annotation Term="Org.OData.Core.V1.Revisions">
+                    <Collection>
+                        <Record>
+                            <PropertyValue Property="Version" String="v1.2" />
+                            <PropertyValue Property="Kind">
+                                <EnumMember>Org.OData.Core.V1.RevisionKind/Added</EnumMember>
+                            </PropertyValue>
+                            <PropertyValue Property="Description" String="desc" />
+                        </Record>
+                    </Collection>
+                </Annotation>
+            </Annotations>
+        </Schema>
+    </edmx:DataServices>
+</edmx:Edmx>
+```
+To add a revision, you must provide the following properties: 
+1. RevisionKind - the kind of revision you are making. Added, Modified or Deprecated.
+1. Description - a description describing the revision you are making.
+1. Version - the version of the model the revision was made. 
```

Comment:
Should all of these be numbered `1`?

Reply:
this is how to number items in .md file..I don't think it will appear like this.

Reply:
![Capture](https://user-images.githubusercontent.com/25525526/131335644-412349ea-9c7f-48f1-be9d-ce841093f60b.JPG)

Reply:
that's how it will appear